classes/external/favourite_package.php

```diff
@@ -0,0 +1,178 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\external;
+
+defined('MOODLE_INTERNAL') || die;
+
+global $CFG;
+require_once($CFG->libdir . "/externallib.php");
+
+use context;
+use context_user;
+use external_api;
+use external_function_parameters;
+use external_value;
+use moodle_exception;
+use qtype_questionpy\utils;
+
+/**
+ * This service can be used to mark a package as favourite.
+ *
+ * @package    qtype_questionpy
+ * @copyright  2024 Jan Britz, TU Berlin, innoCampus - www.questionpy.org
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class favourite_package extends external_api {
+
+    /**
+     * Used to verify the parameters passed to the service.
+     *
+     * @return external_function_parameters
+     */
+    public static function favourite_package_parameters(): external_function_parameters {
+        return new external_function_parameters([
+            'packageid' => new external_value(PARAM_INT),
+            'contextid' => new external_value(PARAM_INT),
```

Comment:
Wieso muss man hier die contextid übergeben? Kannst du nicht über die packageid die contextid bekommen?

Reply:
Die `contextid` ist nicht im Paket selbst hinterlegt, sondern in den einzelnen Paketversionen.

Reply:
Können wir fürs erste so lassen